### ISSUE TITLE
"Package was not restored" error in Xamarin Studio

### ISSUE THREAD

**Issue by paulyoung** [fscheck/FsCheck#116]
I was seeing this error in Xamarin Studio on OS X and figured out that it was due to case-sensitivity through [this question on StackOverflow](http://stackoverflow.com/questions/27502086/cannot-add-linq2twitter-package-to-xamarin-studio/29915300).

After adding FSCheck.NUnit via NuGet in Xamarin Studio my `packages.config` looked like this:
```
<package id="FsCheck.NUnit" version="2.0.3" targetFramework="net45" />
```
However, the following was added to my packages directory:
```
FsCheck.Nunit.2.0.3/FsCheck.Nunit.2.0.3.nupkg`
```

After updating my `packages.config` to:
```
<package id="FsCheck.Nunit" version="2.0.3" targetFramework="net45" />
```
The error went away.

**Comment by paulyoung** [comment]
What's weird is now both the `packages.config` and `packages` directory somehow have the casing of `FsCheck.NUnit`.

I'm not sure how this happened. I think I did another restore or something but definitely didn't rename anything.

**Comment by kurtschelfthout** [comment]
Kind of known nagging problem :/ 

I thought I made all the cases the same as NUnit a while ago but maybe I missed a spot. Or did you perhaps install an older version first?

**Comment by paulyoung** [comment]
I just saw this again after adding things afresh.

I definitely had `packages/FsCheck.Nunit.2.0.4/FsCheck.Nunit.2.0.4.nupkg` and `FsCheck.NUnit` in `packages.config`.

I updated the entry in `packages.config` to `FsCheck.Nunit` and the message went away as before.

However...

I then deleted the `packages` directory and restored all of the packages using the context menu in Xamarin Studio. After that I had `packages/FsCheck.NUnit.2.0.4/FsCheck.NUnit.2.0.4.nupkg` but still had  `FsCheck.Nunit` in `packages.config`, which was now mismatched again.

I reverted the entry `packages.config` to `FsCheck.NUnit` and all was well again.

**Comment by kurtschelfthout** [comment]
So you're saying the case mismatch only happens on the initial install? And package restore does the right thing?

I double checked and I can't find any reference to ``FsCheck.Nunit`` (lowercase) anywhere anymore, but I had those before so it's not unthinkable that some NuGet metadata or package id is confused. May have to ask the NuGet folks to take a look.

**Comment by paulyoung** [comment]
> So you're saying the case mismatch only happens on the initial install? And package restore does the right thing?

I just tried again from scratch and the case mismatch happens on the initial install **but** if all I do is delete the `packages` directory and then restore, it doesn't do the right thing.

It's only when I make the change to `packages.config` and then restore that it makes a difference.

**Comment by kurtschelfthout** [comment]
I can repro this - lower case u on install despite package id having upper case etc - unfortunately nothing in the package has the lower case 'u' so am unable to fix on our side. I'll contact NuGet team to see what can be done.

**Comment by kurtschelfthout** [comment]
The NuGet folks have come back and said this issue is fixed in NuGet 3.0. I've verified that this is the case - at least on windows, visual studio 2015 I get `NUnit` in packages.config, packages/FsCheck.NUnit and the nuget file. 

I don't know if/when nuget 3.0 will make it to Xamarin studio.

They also said they have an open bug item to fix in nuget 2.8.x, and promised to look into trying to fix the FsCheck.NUnit package on the nuget server.

**Comment by kurtschelfthout** [comment]
Closing because old external issue; please re-open if still a problem.
